src/components/Pages/PlaylistCollection/widgets/Playlist/Playlist.js

```diff
@@ -16,6 +16,38 @@ const PlaylistWidget = ({ playlist, selected = 0, link, t }) => {
   const location = useLocation();
 
   const { collection, items, name } = playlist;
+  const { content_type, ccuNames } = collection;
+
+  const getCollectionPartNumber = unitId => {
+    const defaultVal = 1000000;
+    const num = Number(ccuNames[unitId]);
+
+    // put items with not valid collection part numbers at the end
+    return (isNaN(num) || num <= 0) ? defaultVal : num;
+  }
+
+  const sortItems = (item1, item2) => {
+    const val1 = getCollectionPartNumber(item1.unit.id);
+    const val2 = getCollectionPartNumber(item2.unit.id);
+    const result = val1 - val2
+
+    // if equal part number, sort by date and then name
+    if (result === 0) {
+      if (item1.unit.film_date < item2.unit.film_date) {
+        return -1
+      } else if (item1.unit.film_date > item2.unit.film_date) {
+        return 1
+      }
+
+      return item1.unit.name <= item2.unit.name ? -1 : 1;
```

Comment:
If the name is the same you should return 0, right?

Reply:
-1 is fine, i want the same order to remain

Reply:
-1 is not fine. It will force "sort" function to change element places when they are the same.
In some cases it can get to an infinite loop.

If names are the same you should return 0.

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort

Reply:
![image](https://user-images.githubusercontent.com/1877419/169924550-b3e2706d-1ed1-47b4-9dbf-ddbd983b8957.png)